File: merge_two/merge_two.js
// const sortedArray = (arr1, arr2) => {
//   const result = [];
//   let i = 0;
//   let j = 0;

//   while (result.length < arr1.length + arr2.length) {
//     if (arr1[i] === undefined) {
//       result.push(arr2[j])
//       j += 1
//     } else if (arr2[j] === undefined) {
//       result.push(arr1[i])
//       i += 1
//     } else if (arr1[i] < arr2[j]) {
//       result.push(arr1[i])
//       i += 1
//     } else {
//       result.push(arr2[j])
//       j += 1
//     }
//   }
//   return result
// }



const sortedArray = (arr1, arr2) => {
    const result = [];
    let i = 0;
    let j = 0;

    while (result.length < arr1.length + arr2.length) {
        if (arr2[j] === undefined || arr1[i] < arr2[j]) {
            result.push(arr1[i])
            i += 1
        } else {
            result.push(arr2[j])
            j += 1
        }
    }
    return result
}


const arr1 = [1, 1, 1, 1];
const arr2 = [2, 2, 2, 2]
sortedArray(arr1, arr2)

File: merge_two/merge_two.rb
def sortedArray(arr1, arr2)
  result = []
  i = 0
  j = 0

  while result.length < arr1.length + arr2.length 
    if arr2[j] == nil 
      result.push(arr1[i])
      i += 1
    elsif arr1[i] == nil
      result.push(arr2[j])
      j += 1
    elsif 
      arr1[i] < arr2[j]
      result.push(arr1[i])
      i += 1

    else 
      result.push(arr2[j])
      j += 1
    end

  end
  return result
end 


arr1 = [1,3,4,6,8,9,10,14]
arr2 = [1,5,6,8,78,99]
sortedArray(arr1, arr2)